Framework/API/inc/MantidAPI/IFunction.h

```diff
@@ -534,9 +534,9 @@ class MANTID_API_DLL IFunction {
   bool isParallel() const { return m_isParallel; }
 
   /// Set a function handler
-  void setHandler(FunctionHandler *handler);
+  void setHandler(std::shared_ptr<FunctionHandler> handler);
```

Comment:
Could you help me understand why we need to use a `shared_ptr` here?

If we do need one can we use `boost::shared_ptr`? We're using this across the framework and it we be good to keep consistency

Reply:
ok, I wasn't sure if it was best to move to use `std::shared_ptr` since that's available now, or to stick to using boost. 
As for why it has to be shared, I believe that when I went through the code that calls this function I found at least one area where `setHandler` was called on a pointer, and the pointer was returned as well, since it was owned by IFunction it needed to be smart, but I coudln't move a `unique_ptr` in since the place it was called from went on to return that pointer.

Reply:
Thanks for the explanation.

Looking at [`IFunction.cpp`](https://github.com/mantidproject/mantid/blob/master/Framework/API/src/IFunction.cpp#L72) we can see that it deletes `m_handler`, which means it takes ownership of the pointer. I think in this case it would be a good candidate for `setHandler` to accept a `unique_ptr` so that it is clear at the call site that `IFunction` owns it.